### PR TITLE
Quickstart cleanup

### PR DESCRIPTION
**Pull request by chasballew**
@sanderson @fancyremarker 

WHEREAS we want to have a baller support site with clean lines and no cruft,
HEREFORE BE IT RESOLVED, this PR:
- [x] DRYs up article metadata by deprecating front matter in favor of the Middleman data files
- [x] Cleans up page titles and metadata
- [x] Adds data attributes for the Swifttype crawler, for more readable, more relevant results
- [x] Adds a separate Swiftype staging environment

### CODE REVIEW

**Comment by chasballew** [comment]
Comments desired re: testing especially - where/how to test

File: source/layouts/topics.haml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 / Template: topics.haml
 
 = wrap_layout :layout do
-  %header.breadcrumbs
+  %header.breadcrumbs{ 'data-swifttype-index' => 'false' }
```
Comment (skylar-anderson) [inline source/layouts/topics.haml]: @chasballew `data-swiftype-index`

File: source/layouts/quickstart.haml [hunk 2]
```diff
@@ -1,24 +1,22 @@
 / Template: quickstart.haml
 
 = wrap_layout :layout do
-  %header.breadcrumbs
+  %header.breadcrumbs{ 'data-swifttype-index' => 'false' }
```
Comment (skylar-anderson) [inline source/layouts/quickstart.haml]: `data-swiftype-index`

File: source/layouts/layout.haml [hunk 3]
```diff
@@ -37,7 +38,8 @@
                         %i.fa.fa-search
                 #st-results-container
 
-    = yield
+    %div{ 'data-swifttype-name' => 'body', 'data-swiftype-type' => 'text', 'data-swifttype-index' => 'true' }
```
Comment (skylar-anderson) [inline source/layouts/layout.haml]: `data-swiftype-index`  `data-swiftype-name`

**Comment by skylar-anderson** [comment]
@chasballew Nice PR!  Unfortunately it's not very easy to test these Swiftype changes.  One option is to go ahead and deploy to staging and then create a new engine against the staging URL.  

Facebook has a [debugger](https://developers.facebook.com/tools/debug/) for OG tags and twitter has a [debugger](https://cards-dev.twitter.com/validator) for cards, but It doesn't look like Swiftype has any debugger tool other than just initiating a new crawl.

**Comment by skylar-anderson** [comment]
This also reminds me, we need our robots.txt on staging to disallow all so we aren't hit with duplicate content penalties.

**Comment by skylar-anderson** [comment]
Looks like you can add multiple domains for a single engine in Swifttype.  We could add staging, but then set up a rule to blacklist the entire domain.  This would let you see the crawl results before deploying to production but prevent duplicate results.    That being said, I don't think any of these changes will break anything so that might be overkill.

File: config.rb [hunk 4]
```diff
@@ -72,15 +88,43 @@
...
+    og_type = opts[:og_type]
+
+    description ||= current_page.data.header_subtitle
+
+    url = "#{base_url}#{current_page.url}"
+
+    og_type = og_type.nil? ? 'website' : 'article'
 
-    "<title>#{title}</title>"
+    "<meta property=\"og:description\" content=\"#{description}\" >\n" \
```
Comment (fancyremarker) [inline config.rb:133]: :+1:

File: source/partials/_footer.haml [hunk 5]
```diff
@@ -10,9 +10,9 @@
       %li.email
         %a{ href: 'mailto:support@aptible.com' }support@aptible.com
     %ul.nav.nav-pills.footer-nav.pull-right
-      / TODO: Enable documentation section
-      / %li
-      /   %a{ href: '/documentation'}Documentation
+      -# TODO: Enable documentation section
```
Comment (fancyremarker) [inline source/partials/_footer.haml:10]: :+1:

**Comment by chasballew** [comment]
:octopus: posterity octopus: Swiftype recommends using [separate engines for separate environments](https://swiftype.com/documentation/tutorials/rails#managing_multiple).

**Comment by fancyremarker** [comment]
Looks great, @chasballew. Since there are several changes here, can you push to staging for a sanity review?

Also, @sandersonet — yes to the robots.txt disallow on staging. As an additional/alternative safeguard, we can add `rel="canonical"` links on each page that point to the production domain. See [this article](https://support.google.com/webmasters/answer/139066?hl=en) for reference; this should have an equivalent effect to the robots.txt disallow.

**Comment by chasballew** [comment]
screenshot 2015-03-25 17 38 57

New vs old, for Swiftype.

**Comment by chasballew** [comment]
@sandersonet - I think this is ready for final review and merge.

**Pull request merged**